cloudant-http/src/main/java/com/cloudant/http/interceptors/CookieInterceptor.java

```diff
@@ -59,23 +59,28 @@
     private final byte[] sessionRequestBody;
     private final CookieManager cookieManager = new CookieManager();
     private final AtomicBoolean shouldAttemptCookieRequest = new AtomicBoolean(true);
+    private final URL sessionURL;
 
     /**
      * Constructs a cookie interceptor.
-     *
-     * @param username The username to use when getting the cookie
+     *  @param username The username to use when getting the cookie
      * @param password The password to use when getting the cookie
+     * @param baseURL
      */
-    public CookieInterceptor(String username, String password) {
+    public CookieInterceptor(String username, String password, String baseURL) {
 
         try {
+            this.sessionURL = new URL(String.format("%s/_session", baseURL));
             username = URLEncoder.encode(username, "UTF-8");
             password = URLEncoder.encode(password, "UTF-8");
             this.sessionRequestBody = String.format("name=%s&password=%s", username, password)
                     .getBytes("UTF-8");
         } catch (UnsupportedEncodingException e) {
             //all JVMs should support UTF-8, so this should not happen
             throw new RuntimeException(e);
+        } catch (MalformedURLException e) {
```

Comment:
We should move the existing logging for this to here and remove the other `MalformedURLException` catch.